### PR TITLE
A compilation of different branches

### PR DESCRIPTION
**Pull request by prereflect**
A compilation of different branches.

Implement 'search_by_title' on the video model
Implement search in the UI
Recent Videos
Show Only Recent Videos
Implement the front page, register page and sign in page
Add authentication

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
   protect_from_forgery
+
+  helper_method :current_user
+
+  def require_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```
Comment (tomtomecek) [inline app/controllers/application_controller.rb]: This should cache the result of User.find... in an instance variable. Like `@current_user`. This method will be called a lot of times during the request/response so you should use memoization.

File: app/controllers/videos_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class VideosController < ApplicationController
+  before_filter :require_user
```
Comment (tomtomecek) [inline app/controllers/videos_controller.rb]: This should be `before_action`, `_filter` will be deprecated in Rails 5.

File: app/views/shared/_navigation.html.haml [hunk 3]
```diff
@@ -2,37 +2,47 @@
...
+          %li
+            %a{:href => '#'} My Queue
+          %li
+            %a{:href => '#'} People
+        = form_tag search_videos_path, method: 'get',                          |
+                                       class: 'navbar-form navbar-left' do
+          %input.search-query(name='search_term'                               |
+                              type='text'                                      |
+                              class='form-control'                             |
+                              placeholder='Search video by title')             |
```
Comment (tomtomecek) [inline app/views/shared/_navigation.html.haml]: I would rather use `= text_field_tag :search_term, nil, ...` this looks a bit confusing when you mix both HTML/Haml with Rails helpers.

File: app/views/users/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+%section.register.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = bootstrap_form_for @user,
+        layout: :horizontal,
```
Comment (tomtomecek) [inline app/views/users/new.html.haml]: This and 3 lines below are not correctly indented. They are not at the same level as `bootstrap_from_for` anyway.

File: spec/models/video_spec.rb [hunk 5]
```diff
@@ -3,6 +3,32 @@
 describe Video do
   it { should belong_to(:category) }
   it { should validate_presence_of(:title) }
-  it { should validate_uniqueness_of(:title) }
   it { should validate_presence_of(:description) }
+
+  describe 'video search by attribute' do
```
Comment (tomtomecek) [inline spec/models/video_spec.rb]: Please read this guide - http://www.betterspecs.org/#describe on how to properly describe your specs.

File: app/views/users/new.html.haml (hunk 4 above)
Comment (prereflect) [inline app/views/users/new.html.haml]: From [Haml](http://haml.info/docs/yardoc/file.REFERENCE.html#ruby_evaluation)

"A line of Ruby code can be stretched over multiple lines as long as each line but the last ends with a comma."

or am I missing your point @tomtomecek ?

Comment (tomtomecek) [inline app/views/users/new.html.haml]: The options are not equal to the method `bootstrap_form_for`. 

```haml
= bootstrap_form_for @user,
                     layout: :horizontal,
                     label_col: 'col-sm-2',
                     control_col: 'col-sm-4' do |f|
```
Hope its more clear now @prereflect

Comment (prereflect) [inline app/views/users/new.html.haml]: Thank you @tomtomecek

**Pull request merged**